components/push/src/error.rs

```diff
@@ -56,6 +56,10 @@ pub enum ErrorKind {
     /// A failure to parse a URL.
     #[error("URL parse error: {0:?}")]
     UrlParseError(#[from] url::ParseError),
+
+    /// An error if the endpoint could not be fetched
+    #[error("Endpoint Error: {0:?}")]
+    EndpointError(String),
```

Comment:
Minor naming feedback: when I read "endpoint error" my brain parses it as "there was an error with the endpoint", but in this case there was not yet an endpoint at all. Something like "EndpointRegistrationError" may be a better fit in amongst the other names here, although I don't bikeshed the details.

Reply:
I am terrible with naming things - EndpointRegistrationError sounds apt!